soroban-env-host/src/host/declared_size.rs

```diff
@@ -259,6 +264,9 @@ mod test {
         expect!["8"].assert_eq(size_of::<SymbolSmallIter>().to_string().as_str());
         expect!["32"].assert_eq(size_of::<U256>().to_string().as_str());
         expect!["32"].assert_eq(size_of::<I256>().to_string().as_str());
+        #[cfg(target_arch = "x86_64")]
```

Comment:
Interesting to see this! I guess its because [Github actions](https://github.com/stellar/rs-soroban-env/blob/main/.github/workflows/rust.yml#LL57C43-L57C43) is configured to run tests using x86_64 architecture.

Reply:
Yeah, this test is disabled currently. 
It is intended purely as a helper for finding out the proper size to use for the `declared_size` of a new type. It is platform-dependent, the actual `declared_size::<T>` has to be the upper bound of the sizes across all platforms. 
This was causing confusion when someone runs it on a different platform than where the hardcoded value was generated. Thus comes this fix.